src/librustc_metadata/cstore_impl.rs

```diff
@@ -54,81 +62,120 @@ macro_rules! provide {
                 let $cdata = $tcx.sess.cstore.crate_data_as_rc_any($def_id.krate);
                 let $cdata = $cdata.downcast_ref::<cstore::CrateMetadata>()
                     .expect("CrateStore crated ata is not a CrateMetadata");
-                $compute
+                $cdata_fn_compute
+            })*
+
+            $(fn $cnum_fn_name<'a, $lt:$lt>($tcx: TyCtxt<'a, $lt, $lt>, $cnum: CrateNum)
+                        -> <ty::queries::$cnum_fn_name<$lt> as
+                            DepTrackingMapConfig>::Value {
```

Comment:
I think we settled on `DefId`. 

@michaelwoerister can probably tell you how to make a dep-node from a `CrateNum`. I'll try to figure it out later but don't have time for digging right now.